data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
+    for file in not_scaffold_split_yaml_files:
+        print(f"Processing {file}")
+        run_transform(file)
+        ddf = dask.dataframe.read_csv(
+            os.path.join(os.path.dirname(file), "data_clean.csv")
         )
-    else:
-        split_ = pd.DataFrame(
-            np.random.choice(
-                ["train", "test", "valid"],
-                size=len(repr_df),
-                p=[1 - val_size - test_size, test_size, val_size],
+        meta = ("split", "object")  # Meta defines the structure of the new column
+        ddf["split"] = ddf.apply(
+            assign_split,
+            axis=1,
+            meta=meta,
+            args=(test_smiles, val_smiles, train_frac, test_frac, dask_random_state),
+        )
+        split_counts = ddf["split"].value_counts().compute()
+
+        print(
+            f"Dataset {file} has {len(ddf)} datapoints. Split sizes: {split_counts['train']} train, {split_counts['valid']} valid, {split_counts['test']} test."  # noqa: E501
+        )
+
+        # we then write the new data_clean.csv file
+        # if the override option is true, we write this new file to `data_clean.csv` in the same directory
+        # otherwise, we copy the old `data_clean.csv` to `data_clean_old.csv`
+        # and write the new file to `data_clean.csv`
+        if not override:
+            os.rename(
+                os.path.join(os.path.dirname(file), "data_clean.csv"),
+                os.path.join(os.path.dirname(file), "data_clean_old.csv"),
             )
+        ddf.to_csv(
+            os.path.join(os.path.dirname(file), "data_clean.csv"),
+            index=False,
         )
 
-        repr_df["split"] = split_
-        train = repr_df.query("split == 'train'").reset_index(drop=True)
-        test = repr_df.query("split == 'test'").reset_index(drop=True)
-        valid = repr_df.query("split == 'valid'").reset_index(drop=True)
-
-        split = {"train": train, "test": test, "valid": valid}
-
-    # create train and test dataframes
-    train_df = split["train"]
-    test_df = split["test"]
-    valid_df = split["valid"]
-    # add split columns to train and test dataframes
-    train_df["split"] = len(train_df) * ["train"]
-    test_df["split"] = len(test_df) * ["test"]
-    valid_df["split"] = len(valid_df) * ["valid"]
-
-    # merge train and test across all datasets
-    merge = pd.concat([train_df, test_df, valid_df], axis=0)
-    # rewrite data_clean.csv for each dataset
-    rewrite_data_with_splits(
-        csv_paths=paths_to_data,
-        repr_col=repr_col,
-        train_test_df=merge,
-        override=override,
-        check=check,
+
+def scaffold_split(
+    data_dir: Union[str, Path],
+    override: bool = False,
+    train_frac: float = 0.7,
+    val_frac: float = 0.15,
+    test_frac: float = 0.15,
+    seed: int = 42,
+    debug: bool = False,
+):
+    """Performs scaffold splitting on all datasets in the data_dir directory.
```

Comment:
Only on those from the `to_scaffold_split` or am I wrong?

Reply:
yeah. it is the docstring from the function I added yesterday.